scripts/check_e2e_tests_are_captured_in_ci.py

```diff
@@ -0,0 +1,148 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A script to check that travis.yml file & protractor.conf.js have the
+same test suites."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import re
+
+import python_utils
+import utils
+
+# These 4 test suites are not present in travis ci.
+# One is extra ie. (full: [*.js]), and three other tests that
+# are being run by circleCi.
+TEST_SUITES_NOT_RUN_ON_TRAVIS = ['full', 'classroomPage', 'fileUploadFeatures',
+    'topicAndStoryEditor']
+
+
+def get_e2e_suite_names_from_jobs_travis_yml_file():
+    """Extracts the env/jobs section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of names of test suites in sorted alphabetical
+        order of jobs section in travis.yml file.
+    """
+    travis_file_content = read_and_parse_travis_yml_file()
+    jobs_str = python_utils.convert_to_bytes(travis_file_content['env']['jobs'])
+    suites_from_jobs = []
+    # The following line extracts the test suite name from the jobs section
+    # that is in the form RUN_E2E_TESTS_ACCESSIBILITY=true.
+    test_regex = re.compile(r'RUN_E2E_TESTS_([a-zA-Z_-]*)=')
+    jobs = test_regex.findall(jobs_str)
+    for job in jobs:
+        suites_from_jobs.append(
+            utils.snake_case_to_camel_case(job.lower()))
+
+    return sorted(suites_from_jobs)
+
+
+def get_e2e_suite_names_from_script_travis_yml_file():
+    """Extracts the script section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of names of test suites in sorted alphabetical
+        order of script section in travis.yml file.
+    """
+    travis_file_content = read_and_parse_travis_yml_file()
+    script_str = python_utils.convert_to_bytes(travis_file_content['script'])
+    # The following line extracts the test suites from patterns like
+    # bash scripts/run_e2e_tests.sh --suite="accessibility"
+    hyphen_between_regex = re.compile(
+        r'bash scripts/run_e2e_tests.sh --suite="([a-zA-Z_-]*)"')
+    suites_list = hyphen_between_regex.findall(script_str)
+
+    return sorted(suites_list)
+
+
+def get_e2e_suite_names_from_protractor_file():
+    """Extracts the test suites section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all test suites in sorted order
+        in protractor.conf.js file.
+    """
+    protractor_config_file_content = read_protractor_conf_file()
+    # The following line extracts suite object from protractor.conf.js.
+    suite_object_string = re.compile(
+        r'suites = {([^}]+)}').findall(protractor_config_file_content)[0]
+
+    # The following line extracts the keys/test suites from the key: value
+    # pair from the suites object.
+    key_regex = re.compile(r'\b([a-zA-Z_-]*)(?=:)')
```

Comment:
Why do you still need the `?=`? Try to replace that with an explicitly-defined capture group as mentioned above.

Reply:
Done.